atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/atomix/InvalidatingLeaderProxy.java

```diff
@@ -50,13 +52,16 @@ private InvalidatingLeaderProxy(
     public static <T> T create(
             LocalMember localMember,
             DistributedValue<String> leaderId,
+            Election election,
             Supplier<T> delegateSupplier,
             Class<T> interfaceClass) {
         InvalidatingLeaderProxy<T> proxy = new InvalidatingLeaderProxy<>(
                 localMember,
                 leaderId,
                 delegateSupplier);
 
+        election.onElection(term -> proxy.clearDelegateUnchecked());
```

Comment:
@tpetracca The case you describe is possible, though extremely rare. An election is usually only triggered when the leader leaves or is partitioned; but this can happen if the leader is partitioned for just the right amount of time, rejoins for the election and is picked again.

I think at the time @SerialVelocity and I decided that we wouldn't go to great lengths to save this case, though it looks like it might drop out naturally here since we have the term.